index.php

```diff
@@ -10,10 +10,17 @@
     ],
     'hooks' => [
         'page.render:after' => function ($contentType, $data, $html) {
-            if ($contentType === 'html') {
-                $html = str_replace('</body>', snippet('admin-bar', [], true), $html) . '</body>';
+            if ($contentType !== 'html') return;
+
+            // find last occurrence of end tag
+            $pos = strripos($html, '</body>');
+            if ($pos === false) $pos = strripos($html, '</html>');
```

Comment:
The original code would replace the first occurrence of `</body>` in the original HTML. This would fail if:

- The original HTML has the string `</body>` more than once (which is possible if you have some JavaScript strings or HTML in `<script>` or `<template>` elements. Fix: look for the last instance.
- The original HTML used a different case, like `</Body>` or `</BODY>` (uncommon these days, but valid). Fix: use case-insensitive search.

This also falls back to looking for `</html>` if `</body>` was not found. (The result would not be valid HTML but it would still be parsed correctly by browsers.)